src/repositories/repoEntitlementsRepository.ts

```diff
@@ -47,6 +47,21 @@ export class RepoEntitlementsRepository extends BaseRepository {
     }
   }
 
+  async getGatsbyDataSourceByGithubUsername(githubUsername: string) {
+    const query = { github_username: githubUsername };
+    const projection = { _id: 0, gatsby_data_source: 1 };
+    const res = await this.findOne(
+      query,
+      `Mongo Timeout Error: Timedout while retrieving entitlements for ${githubUsername}`,
+      { projection }
+    );
```

Comment:
I'm guessing we're fetching this because we don't know its format right now and this is easiest to change while in development. In the future, if the preview site is as formulaic as I've gleaned, maybe we can save a fetch and cobble the string together. But that's once this is all nailed down.

Reply:
The value here is the site's ID, rather than the name of the preview site. I should probably rename these `gatsby_data_source` variables and fields to make this clearer/more intuitive! Probably `gatsby_site_id` would be more accurate

Reply:
Just made the change. Let me know if this makes more sense!

Reply:
Ah ok, I like that! But definitely my bad for misunderstanding :) sounds good!

Reply:
It's okay! This comment helped me see the light 🙏